tests/semantic/jsonfile/jsonfile.dl

```diff
@@ -0,0 +1,32 @@
+// Souffle - A Datalog Compiler
+// Copyright (c) 2020, The Souffle Developers. All rights reserved
+// Licensed under the Universal Permissive License v 1.0 as shown at:
+// - https://opensource.org/licenses/UPL
+// - <souffle root>/licenses/SOUFFLE-UPL.txt
+// This is a functional test for JSON IO handling.
+// JSON output is directed to file.
+
+.type List = [
+    head : number,
+    tail : List
+]
+.decl A(x : List)
+A(nil).
+A([1,nil]).
+A([2,[3,nil]]).
+.output A(IO=jsonfile,filename="A.json")
+
+.type list = [x:number, y:list]
+.decl B(a:number, b:list)
+
+B(1,[1,[2,[3,nil]]]).
+B(4,nil).
+.output B(IO=jsonfile,filename="B.json")
+
+.decl C(x : List)
+.input C(IO=jsonfile,filename="C.json")
+.output C(IO=json)
+
+.decl D(a:number, b:list)
+.input D(IO=jsonfile,filename="D.json")
+.output D(IO=json)
```

Comment:
Could you add the new line at the end?

We don't really care too deeply about whether there is a line ending at the end of a file, but it is very commonly added in by editors. Eventually either a future dev accidentally makes the change when they didn't mean to, or all future devs have to be careful about reverting changes.  The new line also helps some command line tools give more predictable results.